File: chapter2/insert_sort.rb
# Attempt 1, from basic description of insert sort

def insert_sort list
  (1..list.size-1).each do |index|
    (0..index-1).reverse_each do |reverse_index|
      if list[index] < list[reverse_index]
        tmp = list[index]
        list[index] = list[reverse_index]
        list[reverse_index] = tmp
        index -= 1
      else
        break
      end
    end
  end
  list
end

File: chapter2/insert_sort.ex
defmodule InsertSort do

  def insert_sort(list) do
    InsertSort.insert_sort(list, 1, 0, Enum.at(list, 1) < enum.at(list, 0))
  end

  defp insert_sort(list, index, reverse_index, true) do
  end


  def swap(list, i, j) do
    tmp = Enum.at(list, i)
    List.replace_at(List.replace_at(list, i, Enum.at(list, j)), j, tmp)
  end





end
